clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -73,23 +69,24 @@ abstract class PersistenceDAO(recoveryParallelism: Int) extends LazyLogging {
   /**
     * Write a metadata update to storage.
     *
-    * @param document A partial metadata document representing an upsert.
+    * @param document A JSON object representing an upsert.
     * @param index    Typeclass providing information on where to persist the
     *                 metadata update.
     */
-  def writeUpdate[D <: ClioDocument](
-    document: D,
+  def writeUpdate(
+    document: Json,
     dt: OffsetDateTime = OffsetDateTime.now()
   )(
     implicit ec: ExecutionContext,
-    index: ElasticsearchIndex[D]
+    index: ElasticsearchIndex[_]
   ): Future[Unit] = Future {
-    val jsonString =
-      ModelAutoDerivation.defaultPrinter.pretty(document.asJson(index.encoder))
-
     val writePath = (rootPath / index.persistenceDirForDatetime(dt)).createDirectories()
 
-    val written = (writePath / ClioDocument.persistenceFilename(document.upsertId))
+    val upsertId = new UpsertId(document.findAllByKey(UpsertId.UpsertIdFieldName).head.findAllByKey("id"))
```

Comment:
I am not sure what this constructor call is doing. Also, I see two copies of it. Maybe a named method somewhere (or a constructor override) with a comment that describes what it's doing would be helpful.